torch/cuda/nvtx.py

```diff
@@ -0,0 +1,46 @@
+import os
+import ctypes
+import warnings
+import torch.cuda
+
+lib = None
+
+__all__ = ['range_push', 'range_pop', 'mark']
+
+def _libnvToolsExt():
+    global lib
+    if lib is None:
+        lib = ctypes.cdll.LoadLibrary(None)
+    return lib
+
+def range_push(msg):
+    """
+    Pushes a range onto a stack of nested range span.  Returns zero-based
+    depth of the range that is started.
+
+    Arguments:
+        msg (string): ASCII message to associate with range
+    """
+    if _libnvToolsExt() is None:
+        raise RuntimeError('Unable to load nvToolsExt library')
+    return lib.nvtxRangePushA(ctypes.c_char_p(msg))
+
+def range_pop():
+    """
+    Pops a range off of a stack of nested range spans.  Returns the
+    zero-based depth of the range that is ended.
+    """
+    if _libnvToolsExt() is None:
+        raise RuntimeError('Unable to load nvToolsExt library')
+    return lib.nvtxRangePop()
+
+def mark(msg):
+    """
+    Describe an instantaneous event that occurred at some point.
+
+    Arguments:
+        msg (string): ASCII message to associate with the event.
+    """
+    if _libnvToolsExt() is None:
+        raise RuntimeError('Unable to load nvToolsExt library')
+    return lib.nvtxMarkA(ctypes.c_char_p(msg))
```

Comment:
`nvtxMarkA` returns `void`. I think you may need to set the `restype` in `_libnvToolsExt`. (The default is int -- I'm not sure what happens if you call a void function as if it returns int).